ckan/cli/cli.py

```diff
@@ -39,22 +39,29 @@
 
 class CkanCommand(object):
 
-    def __init__(self, conf=None):
+    def __init__(self, conf=None, setup_logging=True):
         # Don't import `load_config` by itself, rather call it using
         # module so that it can be patched during tests
-        self.config = ckan_cli.load_config(conf)
+        self.config = ckan_cli.load_config(conf, setup_logging=setup_logging)
```

Comment:
```suggestion
        stdout = sys.stdout
        sys.stdout = os.devnull  # prevent logging from affecting cli stdout
        try:
            self.config = ckan_cli.load_config(conf)
        finally:
            sys.stdout = stdout
```
What do you think of this?

No need to check which cli command was issued: we should never be logging to stdout when running the cli. This way correctly configured logging can still work and we don't add a potentially fragile check in the `_init_ckan_config` function.